tests/iface_loopback_action/conftest.py

```diff
@@ -0,0 +1,199 @@
+            rif_port_name = port_dict['port']
+            ports_configuration[rif_port_name] = {}
+            ports_configuration[rif_port_name]['type'] = "ethernet"
+
+        elif index % groups_of_ports == 1:
+            vlan_id = 50 + index
+            rif_port_name = "Vlan{}".format(vlan_id)
+            ports_configuration[rif_port_name] = {}
+            ports_configuration[rif_port_name]['type'] = "vlan"
+            ports_configuration[rif_port_name]['vlan_id'] = vlan_id
+
+        elif index % groups_of_ports == 2:
+            po_id = 50 + index
+            rif_port_name = "PortChannel{}".format(po_id)
+            ports_configuration[rif_port_name] = {}
+            ports_configuration[rif_port_name]['type'] = "po"
+            ports_configuration[rif_port_name]['po_id'] = po_id
+
+        elif index % groups_of_ports == 3:
+            vlan_id = 50 + index
+            rif_port_name = "{}.{}".format(port_dict['port'], vlan_id)
+            ports_configuration[rif_port_name] = {}
+            ports_configuration[rif_port_name]['type'] = "sub_port"
+            ports_configuration[rif_port_name]['vlan_id'] = vlan_id
+
+        elif index % groups_of_ports == 4:
+            po_id = 50 + index
+            vlan_id = 50 + index
+            rif_port_name = "Po{}.{}".format(po_id, vlan_id)
+            ports_configuration[rif_port_name] = {}
+            ports_configuration[rif_port_name]['type'] = "po_sub_port"
+            ports_configuration[rif_port_name]['po_id'] = po_id
+            ports_configuration[rif_port_name]['vlan_id'] = vlan_id
+
+        ports_configuration[rif_port_name]['port'] = port_dict['port']
+        ports_configuration[rif_port_name]['ptf_port'] = port_dict['ptf_port']
+        ports_configuration[rif_port_name]['port_index'] = port_index
+        ports_configuration[rif_port_name]['ip_addr'] = dut_ip_list[index]
+        ports_configuration[rif_port_name]['ptf_ip_addr'] = ptf_ip_list[index]
+        index += 1
+    yield ports_configuration
+
+
+def generate_ip_list():
+    dut_ip_list = []
+    ptf_ip_list = []
+    for i in range(PORT_COUNT):
+        dut_ip_list.append('11.{}.0.1/24'.format(i))
+        ptf_ip_list.append('11.{}.0.10/24'.format(i))
+    return dut_ip_list, ptf_ip_list
+
+
+@pytest.fixture(scope="package", autouse=True)
+def setup(duthost, ptfhost, ori_ports_configuration, ports_configuration, backup_and_restore_config_db_package):
+    """
+    Config: Cleanup the original port configuration and add new configurations before test
+    Cleanup: restore the original configurations
+    :param duthost: DUT host object
+    :param ptfhost: PTF host object
+    :param ori_ports_configuration: original ports configuration parameters
+    :param ports_configuration: ports configuration parameters
+    """
+    remove_ori_dut_port_config(duthost, ori_ports_configuration)
+    apply_config(duthost, ptfhost, ports_configuration)
+    yield
+
+
+
+@pytest.fixture(scope="package", autouse=True)
```

Comment:
why wasn't it merged with 'setup' fixture?

Reply:
> 

it can not be merged, if merged together, then teardown for the fixture backup_and_restore_config_db_package will be executed late then the recover_config, it is not expected.